sphinx/ext/autosummary/generate.py

```diff
@@ -111,6 +112,19 @@ def _underline(title: str, line: str = '=') -> str:
     return title + '\n' + line * len(title)
 
 
+def _is_submodule(module: str, parent: str) -> bool:
+    """Test if specified module name is 'parent' or sub-module of 'parent'"""
+    return module is not None and (parent == module or module.startswith(parent + "."))
+
+
+def _obj_module(o: Any) -> str:
+    """Returns object module name"""
+    if isinstance(o, _types.ModuleType):
+        return o.__name__
+
+    return getattr(o, '__module__', None)
```

Comment:
I think it's safer to use `inspect.getmodule`. For instance, Python built-in singletons such as `have no `__module__` but are still declared in the `builtins` module.